winterfell/controllers/authController.js

```diff
@@ -92,25 +93,96 @@ module.exports = function (app) {
     });
   });
 
-  // Endpoint for routing idme authorization
+  // Server endpoint for redirecting from ID.me authorization
   // https://api.id.me/oauth/authorize?...
   app.get('/auth/link/idme', function(req, res) {
```

Comment:
Cant we just use passport.js for all this? They have a tidy implementation for handling the post redirect, something like:

```
var passport = require('passport')
  , OAuth2Strategy = require('passport-oauth').OAuth2Strategy;

passport.use('provider', new OAuth2Strategy({
    authorizationURL: 'https://www.provider.com/oauth2/authorize',
    tokenURL: 'https://www.provider.com/oauth2/token',
    clientID: '123-456-789',
    clientSecret: 'shhh-its-a-secret'
    callbackURL: 'https://www.example.com/auth/provider/callback'
  },
  function(accessToken, refreshToken, profile, done) {
    User.findOrCreate(..., function(err, user) {
      done(err, user);
    });
  }
));
```

See http://passportjs.org/docs/oauth

Also I noticed from the comment you imply this controller is just for authorization. I think there only needs to be one controller for both authorization and login with OAuth2.0, its all handled on the ID.me side based on what our client id is and whether the user has authorized that client id before.